packages/cursorless-engine/src/languages/LanguageDefinition.ts

```diff
@@ -149,15 +148,13 @@ async function readQueryFileAndImports(
         // if the developer uses the wrong syntax
         /^[^\S\r\n]*;;?[^\S\r\n]*(?:import|include)[^\S\r\n]+['"]?([\w|/.]+)['"]?[^\S\r\n]*$/gm,
         (match) => {
-          const relativeImportPath = match[1];
+          const importName = match[1];
 
           if (doValidation) {
-            validateImportSyntax(ide, queryName, relativeImportPath, match[0]);
+            validateImportSyntax(ide, queryName, importName, match[0]);
           }
 
-          const importQueryPath = join(dirname(queryName), relativeImportPath);
-          rawQueryStrings[importQueryPath] =
-            rawQueryStrings[importQueryPath] ?? null;
+          rawQueryStrings[importName] = rawQueryStrings[importName] ?? null;
```

Comment:
maybe i'm missing something, but haven't you broken relative imports? eg if query is in a subdir, and includes a `..`, that will no longer work. if it's going to be too annoying to support this, i'm not opposed to forcing a flat structure, but might be a bit more future-proof to support subdirs. If we are going to force flat structure, we should prob enforce it tho rather than silently trying to read the wrong file if there is a subdir

Reply:
No relative but folder agnostic is my idea. Just glob all scm files and include by name even if they are in separate folders.

Reply:
that seems a bit unintuitive. also a bit awkward if there is a subdir with several things and everything needs to specify path from root. I don't feel super strongly but if we do it we should prob explicitly catch and throw if we see `../`

alternately, we could expose a `getCanonicalName(base, relative)` function on the storage provider?

Reply:
Not path from root, just name
`import javascript` should work from a parent dir, a subdir etc.
Kinda how npm does it with node modules.

Reply:
but how do you import something in a subdir?

Reply:
No slashes. Just `import foo` it's the providers job to find a query definition with that name. Doesn't matter if you glob files or if they are in an js map. Each definitions name just need to be unique regardless of folder structure.

Reply:
Huh. That would be a bit unexpected. At that point the folders aren't really doing much

Reply:
shall we just force flat for now? i'm not sure it's worth the effort to do anything fancier

Reply:
They are just grouping like all js/ts files in the same folder, but folders have no syntactic meaning.
That was my idea. Or just a flat structure if you prepare.

Reply:
if we do force flat tho would be good to ensure that the node layer is not trying to treat it as a relative path

Reply:
Sounds good

Reply:
I think just an assert here that path.name is equal to path https://github.com/cursorless-dev/cursorless/blob/main/packages/node-common/src/FileSystemLanguageDefinitionsProvider.ts#L41

Reply:
fixed